azext_iot/iothub/_help.py

```diff
@@ -146,3 +146,158 @@ def load_iothub_help():
+        - name: Send cloud-to-device message with custom data and properties.
+          text: >
+            az iot device c2d-message send -d {device_id} -n {iothub_name} --data 'Hello World' --props 'key0=value0;key1=value1'
+        - name: Send a C2D message and wait for device acknowledgement
+          text: >
+            az iot device c2d-message send -d {device_id} -n {iothub_name} --ack full --wait
+    """
+
+    helps[
+        "iot device send-d2c-message"
+    ] = """
+        type: command
+        short-summary: |
+                        Send an mqtt device-to-cloud message.
+                        The command supports sending messages with application and system properties.
+                        Note: If using x509 authentication methods, the certificate and key files must be provided.
+        examples:
+        - name: Basic usage
+          text: az iot device send-d2c-message -n {iothub_name} -d {device_id}
+        - name: Basic usage for device with x509 authentication
+          text: az iot device send-d2c-message -n {iothub_name} -d {device_id} --cp {certificate_file_path} --kp {key_file_path}
+        - name: Basic usage with custom data
+          text: az iot device send-d2c-message -n {iothub_name} -d {device_id} --data {message_body}
+        - name: Send application properties
+          text: az iot device send-d2c-message -n {iothub_name} -d {device_id} --props 'key0=value0;key1=value1'
+        - name: Send system properties (Message Id and Correlation Id)
+          text: az iot device send-d2c-message -n {iothub_name} -d {device_id} --props '$.mid=<id>;$.cid=<id>'
+    """
+
+    helps[
+        "iot device simulate"
+    ] = """
+        type: command
+        short-summary: |
+                      Simulate a device in an Azure IoT Hub.
+                      While the device simulation is running, the device will automatically receive
+                      and acknowledge cloud-to-device (c2d) messages. For mqtt simulation, all c2d messages will
+                      be acknowledged with completion. For http simulation c2d acknowledgement is based on user
+                      selection which can be complete, reject or abandon. The mqtt simulation also supports direct
+                      method invocation which can be acknowledged by a response status code and response payload.
+                      Note: The command by default will set content-type to application/json and content-encoding
+                      to utf-8. This can be overriden.
+                      Note: If using x509 authentication methods, the certificate and key files must be provided.
+        examples:
+        - name: Basic usage (mqtt)
+          text: az iot device simulate -n {iothub_name} -d {device_id}
+        - name: Basic usage for device with x509 authentication (mqtt)
+          text: az iot device simulate -n {iothub_name} -d {device_id} --cp {certificate_file_path} --kp {key_file_path}
```

Comment:
Please add an example with the passphrase parameter

Reply:
added to all (including the dps one)